File: phase_unwrapping/get_flags_charge.m
function CHARGE = get_flags_charge(FLAG)

positive_residue_bit_position = 1;
negative_residue_bit_position = 2;

if bitget(FLAG, positive_residue_bit_position)
    CHARGE = 1;
elseif  bitget(FLAG, negative_residue_bit_position)
    CHARGE = -1;
else
    CHARGE = 0;
end
    

end

File: phase_unwrapping/codegen/mex/remove_dipoles/get_flags_charge.c
/*
 * get_flags_charge.c
 *
 * Code generation for function 'get_flags_charge'
 *
 */

/* Include files */
#include "rt_nonfinite.h"
#include "remove_dipoles.h"
#include "get_flags_charge.h"

/* Function Definitions */
real_T get_flags_charge(uint8_T FLAG)
{
  real_T CHARGE;
  if ((FLAG & 1) != 0) {
    CHARGE = 1.0;
  } else if ((FLAG & 2) != 0) {
    CHARGE = -1.0;
  } else {
    CHARGE = 0.0;
  }

  return CHARGE;
}

/* End of code generation (get_flags_charge.c) */
